packages/core/utils/lib/sanitize/index.js

```diff
@@ -61,10 +61,18 @@ const createContentAPISanitizers = () => {
   };
 
   const sanitizeQuery = async (query, schema, { auth } = {}) => {
-    const { filters, sort, fields, populate } = query;
+    const { filters, sort, fields, populate, ignoreKind } = query;
 
     const sanitizedQuery = cloneDeep(query);
 
+    // This is here since when you do a findMany on single types, it will become a findOne.
+    // To keep it a findMany, you give it the ignoreKind flag.
+    // This sensitization is here since potentially, if a decorator like i18n makes a mistake and returns two values, findOne will only return the first value.
+    // So this query.ignoreKind could return the second record, which is why we sanitize it.
```

Comment:
I think we could be more specific here, wdyt of
```suggestion
    // Some methods behave different if requesting a Content Type or Single Type (entityService findMany)
    // In some cases we want to ignore that behaviour, and we do so by using `ignoreKind` flag.
    // This sanitization is here to avoid potential security issues of this behaviour, specially using decorators & extensions.
```

If we add more behaviours like this in other places (I hope not) this comment would be generic enough to understand the problem imo.